### PR TITLE
Combining relationships and metadata during submission

### PR DESCRIPTION
**Pull request by LotteHofstede**
This PR is dependent on #534.

It introduces the possibility to add relationships and metadata values to the same field during submission. These different types of values can be reordered using a drag icon.

### CODE REVIEW

**Comment by tdonohue** [comment]
@LotteHofstede : Just a note that this PR needs updating/rebasing now that #534 is merged.

File: src/app/submission/objects/submission-objects.effects.ts [hunk 1]
```diff
@@ -136,7 +136,7 @@ export class SubmissionObjectEffects {
     ofType(SubmissionObjectActionTypes.SAVE_SUBMISSION_FORM_SUCCESS, SubmissionObjectActionTypes.SAVE_SUBMISSION_SECTION_FORM_SUCCESS),
     withLatestFrom(this.store$),
     map(([action, currentState]: [SaveSubmissionFormSuccessAction | SaveSubmissionSectionFormSuccessAction, any]) => {
-      return this.parseSaveResponse((currentState.submission as SubmissionState).objects[action.payload.submissionId], action.payload.submissionObject, action.payload.submissionId);
+      return this.parseSaveResponse((currentState.submission as SubmissionState).objects[action.payload.submissionId], action.payload.submissionObject, action.payload.submissionId, action.payload.notify);
```
Comment (paulo-graca) [inline src/app/submission/objects/submission-objects.effects.ts]: Not related with this specific change, but it's in this class, I'm getting this blocking error when accessing the submission form and I can't test this PR:

```
ERROR TypeError: Cannot read property 'forEach' of undefined
    at MapSubscriber.project (0.client.js:1282)
    at MapSubscriber../node_modules/rxjs/_esm5/internal/operators/map.js.MapSubscriber._next (client.js:196768)
    at MapSubscriber../node_modules/rxjs/_esm5/internal/Subscriber.js.Subscriber.next (client.js:190809)
    at FilterSubscriber../node_modules/rxjs/_esm5/internal/operators/filter.js.FilterSubscriber._next (client.js:196213)
    at FilterSubscriber../node_modules/rxjs/_esm5/internal/Subscriber.js.Subscriber.next (client.js:190809)
    at ScannedActionsSubject../node_modules/rxjs/_esm5/internal/Subject.js.Subject.next (client.js:190575)
    at SafeSubscriber._next (client.js:112368)
    at SafeSubscriber../node_modules/rxjs/_esm5/internal/Subscriber.js.SafeSubscriber.__tryOrUnsub (client.js:190950)
    at SafeSubscriber../node_modules/rxjs/_esm5/internal/Subscriber.js.SafeSubscriber.next (client.js:190888)
    at Subscriber../node_modules/rxjs/_esm5/internal/Subscriber.js.Subscriber._next (client.js:190832)
```

It's at line 59 - https://github.com/DSpace/dspace-angular/blob/32f8842af8c2e67ca7076221eef241df4b008b31/src/app/submission/objects/submission-objects.effects.ts#L59

**Comment by benbosman** [comment]
I've tested this UI against a REST server containing:
* https://github.com/DSpace/DSpace/pull/2691
* https://github.com/DSpace/DSpace/pull/2746

In order to test various types of submissions, I've configured:
* [submission-forms.xml](https://github.com/DSpace/dspace-angular/files/4556752/submission-forms.xml.zip) to contain:
  * In the form traditionalpageone, relation-field isAuthorOfPublication, I've set `name-variants` to true
  * In the form traditionalpageone, I've created a relation-field isOrgUnitOfPublication
  * In the form orgUnitStep, I've created a relation-field isParentOrgUnitOf and isChildOrgUnitOf
* item-submission.xml to contain a collection mapping, e.g.:
  * `<name-map collection-handle="123456789/6" submission-name="People"/>`
  * `<name-map collection-handle="123456789/8" submission-name="OrgUnit"/>`
  * `<name-map collection-handle="123456789/29" submission-name="JournalVolumes"/>`
* virtual-metadata.xml to contain in `publicationOrgUnit_name`:
  * `<property name="useForPlace" value="true"/>`
  * `<property name="populateWithNameVariant" value="false"/>`

Using this setup, I've tested:
* Submitting a [publication](https://dspace7-entities.atmire.com/items/f6dd0d1d-20cf-43d3-9bb4-be5e861964bb) with a [video](https://www.dropbox.com/s/jl6qhxvlszsgk0s/Schermopname%202020-04-30%20om%2009.34.46.mov?dl=0):
  * using the lookup to add an author entity
  * using the lookup to add an author entity with a name variant
  * using the lookup to add an author entity with a name variant also stored on the related item
  * adding a plain text author
  * moving between plain text authors and related entities
  * using the lookup to add an org unit entity which is configured not to allow name variants
  * adding a plain text org unit
  * moving between plain text org unit and related entities
  * The item display for publications has not been adjusted to display plain text org units, but you can see the order on the full item page
* Submitting an [org unit](https://dspace7-entities.atmire.com/items/0b8025d7-3071-4980-b528-8bb7bef8cf4e) with a [video](https://www.dropbox.com/s/tkvy49jdlesnk33/Schermopname%202020-04-30%20om%2009.44.55.mov?dl=0):
  * using the lookup to add a parent org unit without name variants, not repeatable (each child can only have one parent), without support for plain text
  * using the lookup to add a child org unit without name variants, repeatable, without support for plain text
  * trying to add a child org unit which already has a parent, and which is refused to create a second parent relationship
  * The item display for org units has not been adjusted to display parent/child org units, but you can see the relationships on the full item page

What I can see as open problems so far:
* When trying to create a relationship which is not supported, you can see REST returns an error, but Angular doesn't display the error. It just didn't add the relationship in the background to the "Child org unit" section. It did select the checkbox and kept it selected even though the relationship could not be created. This is not a blocking issue for me, and can be solved in a separate PR

**Comment by artlowel** [comment]
Thanks for the review @benbosman I'll see what I can do.

For clarity's sake @paulo-graca & @tdonohue this PR should now be ready for review.

**Comment by paulo-graca** [comment]
I've just tested against master branch and it doesn't work. I will try it also with #2691.

**Comment by paulo-graca** [comment]
I was expecting to have a different result but I could not. Sorry!

The test I did on the UI:

- Start a new submission with a specific collection

- On the default form, for authors, I clicked on "Lookup". But I noticed that I was getting an error on dspace.log
```
2020-04-30 18:30:57,161 WARN  org.dspace.discovery.SolrServiceImpl @ Not able to retrieve object RESOURCE_ID:665f40e2-242a-4c24-814c-bfc96737741f - RESOURCE_TYPE_ID:Item
2020-04-30 18:30:57,161 ERROR org.dspace.discovery.SolrServiceImpl @ MY_USER::Error while retrieving DSpace object from discovery index:Unique identifier\colon; null
```
That error was related with entity deletion PR tests that didn't sent data to solr. I removed the solr document manually and I didn't got any more dspace errors (this might be an issue).

- Then, firstly I tried to use Orcid to import from a external source. But, besides getting the total records, I didn't see any results. So I picked an already existent entity from local authors. 

- I noticed the current selection had the correct entity I picked, then I closed the form. I noticed that nothing happens and also didn't got any errors.

- I also tried to add metadata (author's name) but couldn't see the picked entity.

For these tests I used master, atmire:w2p-68820_Moving-relationships-ITs and atmire:w2p-68820_Moving-relationships-ITs + atmire:circular-entities

**Comment by artlowel** [comment]
@paulo-graca Thanks for testing

I was able to reproduce your ORCID issue, but it also happens on master. It's not an issue with this PR.

This PR focusses on support for adding preexisting local entities in fields that also support plain metadata values, and rewriting the reordering code to support those mixed fields, and have better usability in general.

**Comment by paulo-graca** [comment]
> I was able to reproduce your ORCID issue, but it also happens on master. It's not an issue with this PR.

Thank you Art. I understood that. That's why I tested with a local entity. But was unable to have any effect on the form itself.

**Comment by artlowel** [comment]
Strange.

Would you mind testing against https://dspace7-entities.atmire.com/server to eliminate the rest api as a variable? The default accounts should work again, with their expected password.

**Comment by paulo-graca** [comment]
Sure I will try with your backend. 
This is what is occurring:

1. Form loading and picking a local entity
image

2. Confirming the user is selected
image

3. After closing the form, nothing happens.
image

**Comment by paulo-graca** [comment]
Using your DB/rest, I'm getting the same result with an issue regarding the "current selection":
1. Choose an entity
image

2. Current selection (not showing any results):
image

3. Back to the form
image

**Comment by artlowel** [comment]
I don't know what's going wrong for you 

I just checked out this branch in a new folder, built it in prod mode, tried it again against the https://dspace7-entities.atmire.com/server and it works:

out2

**Comment by benbosman** [comment]
@paulo-graca how are you choosing the collection for your submission?
The collection chooser in the submission is currently still broken, and doesn't apply item templates yet. It's important to start the submission immediately in a collection which has the relationship type set correctly. Or alternatively setting the relationship type as a submission field.
The new item functionality in the navigation is currently the only way a submission can be started with an item template

**Comment by benbosman** [comment]
@artlowel I noticed one more issue with the qual dropdowns, but it also behaves strange in master.
If you enter multiple values in the qual dropdown, you can loose a value, as visible in https://www.dropbox.com/s/ngte7uvugd00xu0/Schermopname%202020-05-07%20om%2010.28.32.mov?dl=0
The data sent to REST is:
```json
[{"op":"add","path":"/sections/traditionalpageone/dc.identifier.issn","value":[{"value":"ISSN 1","language":null,"authority":null,"display":"ISSN 1","confidence":-1,"place":0,"otherInformation":null}]},{"op":"add","path":"/sections/traditionalpageone/dc.identifier.issn/-","value":{"value":"ISSN 2","language":null,"authority":null,"display":"ISSN 2","confidence":-1,"place":0,"otherInformation":null}}]
```

In master, it sends the first ISSN twice to the server, overwriting the full ISSN contents in the second request (also rather strange behavior):
```json
[{"op":"add","path":"/sections/traditionalpageone/dc.identifier.issn","value":[{"value":"ISSN 1","language":null,"authority":null,"display":"ISSN 1","confidence":-1,"place":0,"otherInformation":null}]},{"op":"add","path":"/sections/traditionalpageone/dc.identifier.issn","value":[{"value":"ISSN 1","language":null,"authority":null,"display":"ISSN 1","confidence":-1,"place":0,"otherInformation":null},{"value":"ISSN 2","language":null,"authority":null,"display":"ISSN 2","confidence":-1,"place":0,"otherInformation":null}]},{"op":"add","path":"/sections/traditionalpageone/dc.identifier.ismn","value":[{"value":"ISMN 1","language":null,"authority":null,"display":"ISMN 1","confidence":-1,"place":0,"otherInformation":null}]}]
```

**Comment by paulo-graca** [comment]
Thank you Ben for the tip! I was able to have it working.
image

I'm going to try it out with my local  database and give more feedback. Just a note. This PR is now far behind the master branch.

**Comment by paulo-graca** [comment]
image

It worked as expected. As @benbosman referred I needed a pre-defined itemTemplate for the collection with a defined `relationship.type`. Perhaps it makes some sense to assume if none defined, use Publication, or have some default behavior if relationship.type not available (not on the scope of this PR). 
I will do some more tests, but this PR is now far behind master. Since there are more recent changes and some code conflicts I need to come back to this PR and repeat my tests. 

Thank you!

**Comment by paulo-graca** [comment]
I could repeat my test after code update:
image

File: src/app/core/submission/workspaceitem-data.service.ts [hunk 2]
```diff
@@ -21,7 +21,7 @@ import { WorkspaceItem } from './models/workspaceitem.model';
 @dataService(WorkspaceItem.type)
 export class WorkspaceitemDataService extends DataService<WorkspaceItem> {
   protected linkPath = 'workspaceitems';
-  protected responseMsToLive = 10 * 1000;
+  // protected responseMsToLive = 10 * 1000;
```
Comment (paulo-graca) [inline src/app/core/submission/workspaceitem-data.service.ts:24]: do we need to left this comment?

File: src/app/entity-groups/research-entities/submission/item-list-elements/person/person-search-result-list-submission-element.component.html [hunk 3]
```diff
@@ -1,7 +1,7 @@
 <div class="d-flex">
-    <div class="person-thumbnail pr-2">
-        <ds-thumbnail [thumbnail]="getThumbnail() | async" [defaultImage]="'assets/images/person-placeholder.svg'"></ds-thumbnail>
-    </div>
+<!--    <div class="person-thumbnail pr-2">-->
+<!--        <ds-thumbnail [thumbnail]="getThumbnail() | async" [defaultImage]="'assets/images/person-placeholder.svg'"></ds-thumbnail>-->
+<!--    </div>-->
```
Comment (paulo-graca) [inline src/app/entity-groups/research-entities/submission/item-list-elements/person/person-search-result-list-submission-element.component.html:4]: do we need to left this comment?

**Comment by paulo-graca** [comment]
I was doing some more tests and I think I've just found an issue with this PR.
image

When trying to create a new collection or community (from admin menu), I'm getting the 404 component (when this occurs, no request are made). 
I've just tested the same with master branch and the same doesn't happen with it.

**Comment by benbosman** [comment]
@artlowel while doing some random tests on this, I noticed a problem with the hints.
In master: https://www.dropbox.com/s/kaxopf9ay89n18v/Schermafdruk%202020-06-29%2012.08.26.png?dl=1
In this branch: https://www.dropbox.com/s/sdigdy8l2uhoblz/Schermafdruk%202020-06-29%2012.08.55.png?dl=1

This is the same REST backend, but not all hints are displayed anymore

**Comment by artlowel** [comment]
@benbosman those hints should be fixed now

@paulo-graca Is it possible that the community & collection issue happened when you started the UI in dev mode first, then ran `yarn run test` while leaving the dev server on, and then tried to create a community or collection and got that error? 

Because I've seen that before on another branch. Since the switch to angular cli there is only one active environment file for dev, test and prod mode, and it is swapped when you start one of these. If you overwrite the dev environment with the test environment you get this issue because the context path changes. We should look in to a workaround for that issue because it is quite annoying.

In any case, if I just check out this branch, build it and test it, I can't reproduce your issue

**Comment by benbosman** [comment]
Thanks @artlowel the hints are also displayed correctly in my test setup now

**Comment by paulo-graca** [comment]
> @paulo-graca Is it possible that the community & collection issue happened when you started the UI in dev mode first, then ran `yarn run test` while leaving the dev server on, and then tried to create a community or collection and got that error?
> 
Thank you @artlowel!  But it seems not be that case. 

I've only ran once -  `yarn run start:dev`

```
# ps auxf|grep node
root     26431  0.1  0.7 882600 27816 pts/0    Sl+  12:36   0:00  |       \_ node /root/.yarn/bin/yarn.js run start:dev
root     26484  0.0  0.2 597388 10424 pts/0    Sl+  12:36   0:00  |           \_ /usr/bin/node /usr/local/dspace/dspace-angular541_2_git/node_modules/.bin/npm-run-all --parallel config:dev:watch serve
root     26495  0.1  0.7 882372 29192 pts/0    Sl+  12:36   0:00  |               \_ /usr/bin/node /root/.yarn/bin/yarn.js run config:dev:watch
root     26524  0.0  0.2 794184 11052 pts/0    Sl+  12:36   0:00  |               |   \_ /usr/bin/node /usr/local/dspace/dspace-angular541_2_git/node_modules/.bin/nodemon
root     26496  0.1  0.7 882352 29232 pts/0    Sl+  12:36   0:00  |               \_ /usr/bin/node /root/.yarn/bin/yarn.js run serve
root     26513  0.7  2.5 665184 100472 pts/0   Sl+  12:36   0:04  |                   \_ /usr/bin/node /usr/local/dspace/dspace-angular541_2_git/node_modules/.bin/ts-node --project ./tsconfig.ts-node.json scripts/serve.ts
root     26562  0.1  1.3 615036 50368 pts/0    Sl+  12:36   0:01  |                           \_ /usr/bin/node /usr/local/dspace/dspace-angular541_2_git/node_modules/@ngtools/webpack/src/type_checker_worker.js 9d93e901-158a-4cf9-ba1b
```

I've started the process from scratch, again the same occurs. Probably can be related with my data (like the requirement fora  missing relationship.type). I don't see any requests or errors. I will test it with your server.

**Comment by paulo-graca** [comment]
I was able to test it with Atmire's server (thank you @benbosman for adding my server!) and the result is the same.
404-error

**Comment by artlowel** [comment]
Ok, I'll look in to it further. Thanks for testing!

**Comment by paulo-graca** [comment]
Ok thank you @artlowel! This is what I'm getting with master branch, with same server configuration.
non-404-error

**Comment by artlowel** [comment]
@paulo-graca I was able to reproduce your issue, and I think I fixed it.

**Comment by paulo-graca** [comment]
I got this compiling error:

```
ERROR in src/app/core/data/item-template-data.service.ts:124:5 - error TS2322: Type 'Observable<RestResponse>' is not assignable to type 'Observable<boolean>'.
  Type 'RestResponse' is not assignable to type 'boolean'.

124     return super.delete(item.uuid);
```

**Comment by artlowel** [comment]
@paulo-graca I'm sorry. That should be fixed now

**Comment by tdonohue** [comment]
This PR needs a minor rebase.  Could we also resolve the LGTM alerts noted [here](https://github.com/DSpace/dspace-angular/pull/541#issuecomment-657811662)?  It's showing that, while this PR fixes some alerts, it creates 32 new minor issues (mostly unused variables it seems)

File: src/app/shared/form/builder/ds-dynamic-form-ui/ds-dynamic-form-control-container.component.ts [hunk 4]
```diff
@@ -351,39 +376,47 @@ export class DsDynamicFormControlContainerComponent extends DynamicFormControlCo
...
-            return observableOf(undefined) as Observable<RemoteData<Relationship>>;
-          }
-        })
-      ).subscribe();
-    })
+  onRemove(): void {
+    const arrayContext: DynamicFormArrayModel = (this.context as DynamicFormArrayGroupModel).context;
+    const path = this.formBuilderService.getPath(arrayContext);
+    const formArrayControl = this.group.root.get(path) as FormArray;
+    console.log('this.listId', this.listId);
```
Comment (tdonohue) [inline src/app/shared/form/builder/ds-dynamic-form-ui/ds-dynamic-form-control-container.component.ts:418]: Please remove this logging statement (or comment out)

File: src/app/shared/form/builder/ds-dynamic-form-ui/models/scrollable-dropdown/dynamic-scrollable-dropdown.component.ts [hunk 5]
```diff
@@ -119,6 +119,7 @@ export class DsDynamicScrollableDropdownComponent extends DynamicFormControlComp
 
   onSelect(event) {
     this.group.markAsDirty();
+    console.log('onSelect event', event);
```
Comment (tdonohue) [inline src/app/shared/form/builder/ds-dynamic-form-ui/models/scrollable-dropdown/dynamic-scrollable-dropdown.component.ts:122]: Please remove this logging statement (or comment out)

File: src/app/shared/form/builder/ds-dynamic-form-ui/relation-lookup-modal/relationship.effects.ts [hunk 6]
```diff
@@ -80,33 +94,58 @@ export class RelationshipEffects {
...
+              this.updateAfterPatchSubmissionId = submissionId;
+              this.relationshipService.refreshRelationshipItemsInCache(item1);
+              this.relationshipService.refreshRelationshipItemsInCache(item2);
+            });
           }
         }
       )
     );
 
+  @Effect({ dispatch: false }) updateRelationshipActions$ = this.actions$
```
Comment (tdonohue) [inline src/app/shared/form/builder/ds-dynamic-form-ui/relation-lookup-modal/relationship.effects.ts:118]: Could you add some basic TypeDocs to this new Effect, similar to how other Effects in this same file are described?

File: src/app/shared/form/builder/ds-dynamic-form-ui/relation-lookup-modal/relationship.effects.ts [hunk 7]
```diff
@@ -80,33 +94,58 @@ export class RelationshipEffects {
...
+    .pipe(
+      ofType(RelationshipActionTypes.UPDATE_RELATIONSHIP),
+      map((action: UpdateRelationshipAction) => {
+        this.updateAfterPatchSubmissionId = action.payload.submissionId;
+      })
+    );
+
+  @Effect() commitServerSyncBuffer = this.actions$
```
Comment (tdonohue) [inline src/app/shared/form/builder/ds-dynamic-form-ui/relation-lookup-modal/relationship.effects.ts:126]: Same here, I'd appreciate some TypeDocs / comments to describe this Effect.

File: src/app/shared/form/builder/ds-dynamic-form-ui/relation-lookup-modal/relationship.effects.ts [hunk 8]
```diff
@@ -119,17 +158,36 @@ export class RelationshipEffects {
...
-      take(1)
-    ).subscribe();
+      take(1),
+      switchMap(() => this.refreshWorkspaceItemInCache(submissionId)),
+    ).subscribe((submissionObject: SubmissionObject) => {
+      this.store.dispatch(new SaveSubmissionSectionFormSuccessAction(submissionId, [submissionObject], false))
+    });
+  }
+
+  refreshWorkspaceItemInCache(submissionId: string): Observable<SubmissionObject> {
```
Comment (tdonohue) [inline src/app/shared/form/builder/ds-dynamic-form-ui/relation-lookup-modal/relationship.effects.ts:177]: Please add TypeDocs to describe this new method.

File: src/app/shared/form/builder/ds-dynamic-form-ui/relation-lookup-modal/selection-tab/dynamic-lookup-relation-selection-tab.component.ts [hunk 9]
```diff
@@ -29,10 +29,7 @@ import { createSuccessfulRemoteDataObject } from '../../../../../remote-data.uti
  * Tab for inside the lookup model that represents the currently selected relationships
  */
 export class DsDynamicLookupRelationSelectionTabComponent {
-  /**
-   * The label to use to display i18n messages (describing the type of relationship)
-   */
-  @Input() label: string;
+  @Input() relationshipType: string;
```
Comment (tdonohue) [inline src/app/shared/form/builder/ds-dynamic-form-ui/relation-lookup-modal/selection-tab/dynamic-lookup-relation-selection-tab.component.ts:32]: Could you add a comment to describe this new Input, similar to how others are described?  It looks like a comment existed previously, but you accidentally removed it.

File: src/app/submission/objects/submission-objects.effects.ts [hunk 10]
```diff
@@ -232,18 +242,45 @@ export class SubmissionObjectEffects {
         catchError(() => observableOf(new DiscardSubmissionErrorAction(action.payload.submissionId))));
     }));
 
+  @Effect() addAllMetadataToSectionData = this.actions$.pipe(
```
Comment (tdonohue) [inline src/app/submission/objects/submission-objects.effects.ts:245]: Please add a comment/TypeDocs to describe this new Effect.

File: src/app/shared/form/builder/ds-dynamic-form-ui/relation-lookup-modal/relationship.effects.ts (hunk 8 above)
Comment (LotteHofstede) [reply src/app/shared/form/builder/ds-dynamic-form-ui/relation-lookup-modal/relationship.effects.ts]: Added TypeDoc and made method private

File: src/app/shared/form/builder/ds-dynamic-form-ui/relation-lookup-modal/relationship.effects.ts (hunk 7 above)
Comment (LotteHofstede) [reply src/app/shared/form/builder/ds-dynamic-form-ui/relation-lookup-modal/relationship.effects.ts]: Added TypeDoc

File: src/app/shared/form/builder/ds-dynamic-form-ui/relation-lookup-modal/relationship.effects.ts (hunk 6 above)
Comment (LotteHofstede) [reply src/app/shared/form/builder/ds-dynamic-form-ui/relation-lookup-modal/relationship.effects.ts:121]: Added TypeDoc

File: src/app/shared/form/builder/ds-dynamic-form-ui/models/scrollable-dropdown/dynamic-scrollable-dropdown.component.ts (hunk 5 above)
Comment (LotteHofstede) [reply src/app/shared/form/builder/ds-dynamic-form-ui/models/scrollable-dropdown/dynamic-scrollable-dropdown.component.ts]: Added TypeDoc

File: src/app/shared/form/builder/ds-dynamic-form-ui/ds-dynamic-form-control-container.component.ts (hunk 4 above)
Comment (LotteHofstede) [reply src/app/shared/form/builder/ds-dynamic-form-ui/ds-dynamic-form-control-container.component.ts]: Removed console.log

File: src/app/shared/form/builder/ds-dynamic-form-ui/relation-lookup-modal/selection-tab/dynamic-lookup-relation-selection-tab.component.ts (hunk 9 above)
Comment (LotteHofstede) [reply src/app/shared/form/builder/ds-dynamic-form-ui/relation-lookup-modal/selection-tab/dynamic-lookup-relation-selection-tab.component.ts:35]: Added TypeDoc

File: src/app/submission/objects/submission-objects.effects.ts (hunk 10 above)
Comment (LotteHofstede) [reply src/app/submission/objects/submission-objects.effects.ts:248]: Added TypeDoc

**Comment by LotteHofstede** [comment]
@tdonohue Thanks for the additional feedback, I removed the console.log and added the requested TypeDoc!

**Comment by tdonohue** [comment]
Thanks @LotteHofstede !  This is at +2 & it fixes a community reported bug, so merging immediately.

**Pull request merged**